### DISCUSSION TITLE
GCSFileTransformOperator not rendering any template variable

### DISCUSSION THREAD

**Discussion by rlebrao** [apache/airflow#19629] [Q&A] [answered]
I'm trying to use {{ ds }}, or any template variable, in the **source_object** and **destination_object** param, but I was getting a 404 error for the bucket I was looking for. In the logs, I can see that this error is because the variable was not rendered.

I'm currently using Airflow v2.1.0.

Anyone with the same problem?

**Comment by potiuk** [comment]
Those paraemeters are not in the "template_fields" of the operator:

```
    template_fields = (
        'source_bucket',
        'destination_bucket',
        'transform_script',
        'impersonation_chain',
    )
```

You can create your own custom operator deriving from the `GCSFileTransformOperator` and override template_fields in it. Something like that:

```
class MyGCSFileTransformOperator(GCSFileTransformOperator):
    template_fields = (
        'source_object',
        'destination_object',
        'source_bucket',
        'destination_bucket',
        'transform_script',
        'impersonation_chain',
    )
```

Then you could even submit a PR to Airlfow to add it to the original operator.

**Reply by rlebrao** [reply]
Hi! Thanks for the quick response.

I'm being able to use the same template variables with the LocalFilesystemToGCSOperator (https://airflow.apache.org/docs/apache-airflow-providers-google/1.0.0/_api/airflow/providers/google/cloud/transfers/local_to_gcs/index.html)

Even though the docs I'm only seeing these template fields: 
template_fields= ['src', 'dst', 'bucket', 'impersonation_chain']

How would this be possible? I thought the default template variable should work in any operator: https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html

**Reply by potiuk** [reply]
Just follow the JinjaTemplating link and read it: https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html#concepts-jinja-templating

**Reply by rlebrao** [reply]
Sure, I read it, but still not exactly my point 🙁 
This is the code where I declare the operator, I'm using the {{ds}} variable, as in the example.
```
    exec_date = "{{ds}}"
    reports_transform = GCSFileTransformOperator(
        task_id="reports_transform",
        source_bucket =BUCKET,
        source_object=f"reports/{exec_date}/reports_report_raw.csv",
        destination_bucket = BUCKET,
        destination_object = f"reports/{exec_date}/reports_report_normalized.csv",
        transform_script=['python',f'{CUR_DIR}/scripts/normalize_columns.py'],
    )
```

This is the error message I'm getting
```
ERROR - Download attempt of object: reports/{{ds}}/reports_report_raw.csv from reports/{{ds}}/reports_report_raw.csv has failed. Attempt: 1, max 1.
```
In the message, I can see that the variable was not rendered

**Reply by potiuk** [reply]
I answered above - it's not rendered because 'source_object' is not in 'template_fields'. Just read the link

**Comment by rlebrao** [comment, marked as answer]
Closing it. I was trying to use the variable in a non templated field
